Documentation/project-docs/AssemblyBlobs.md

```diff
@@ -0,0 +1,214 @@
+
+Assemblies are stored as adjacent byte streams:
+
+ - **Image data**
+   Required to be present for all assemblies, contains the actual
+   assembly PE image.
+ - **Debug data**
+   Optional. Contains the assembly's PDB or MDB debug data.
+ - **Config data**
+   Optional. Contains the assembly's .config file. Config data
+   **must** be terminated with a `NUL` character (`0`), this is to
+   make runtime code slightly more efficient.
+
+All the structures described here are defined in the
+[`xamarin-app.hh`](../../src/monodroid/jni/xamarin-app.hh) file.
+Should there be any difference between this document and the
+structures in the header file, the information from the header is the
+one that should be trusted.
+
+## Common header
+
+All kinds of blobs share the following header format:
+
+    struct BundledAssemblyBlobHeader
+    {
+        uint32_t magic;
+        uint32_t version;
+        uint32_t local_entry_count;
+        uint32_t global_entry_count;
+        uint32_t blob_id;
+    ;
+
+Individual fields have the following meanings:
+
+ - `magic`: has the value of 0x41424158 (`XABA`)
+ - `version`: a value increased every time blob format changes.
+ - `local_entry_count`: number of assemblies stored in this blob (also
+   the number of entries in the assembly descriptor table, see below)
+ - `global_entry_count`: number of entries in the index blob's (see
+   below) hash tables, all the other blobs store `0` in this field
+ - `blob_id`: a unique ID of this blob.
+ 
+## Assembly descriptor table
+
+Each blob header is followed by a table of
+`BundledAssemblyBlobHeader.local_entry_count` entries, each entry
+defined by the following structure:
+
+    struct BlobBundledAssembly
+    {
+        uint32_t data_offset;
+        uint32_t data_size;
+        uint32_t debug_data_offset;
+        uint32_t debug_data_size;
+        uint32_t config_data_offset;
+        uint32_t config_data_size;
+    };
+
+Only the `data_offset` and `data_size` fields must have a non-zero
+value, other fields describe optional data and can be set to `0`. 
+
+Individual fields have the following meanings:
+
+  - `data_offset`: offset of the assembly image data from the
+    beginning of the blob file
+  - `data_size`: number of bytes of the image data
+  - `debug_data_offset`: offset of the assembly's debug data from the
+    beginning of the blob file. A value of `0` indicates there's no
+    debug data for this assembly.
+  - `debug_data_size`: number of bytes of debug data. Can be `0` only
+    if `debug_data_offset` is `0`
+  - `config_data_offset`: offset of the assembly's config file data
+    from the  beginning of the blob file. A value of `0` indicates
+    there's no config file data for this assembly.
+  - `config_data_size`: number of bytes of config file data. Can be
+    `0` only if `config_data_offset` is `0`
+
+## Index blob
+
+Each application will contain exactly one blob with a global index -
```

Comment:
This concerns me; https://github.com/xamarin/xamarin-android/pull/6311/files#r725237133

Reply:
I don't think it's going to be a problem